lib/SILOptimizer/Mandatory/TFLowerGraph.cpp

```diff
@@ -815,6 +826,89 @@ static TF_Tensor *convertValuesToTensor(ArrayRef<APInt> elts,
   return tensor;
 }
 
+TF_Output TFGraphLowering::createUndefNode(SILType type) {
+  // Create some magic constant. Actual value does not matter as it will
+  // not be used in any calculations.
+  // FIXME: Is there a better way to model undefs in TF Graph if we
+  // don't have a way of getting rid of undefs during canonicalization.
+  APInt value;
+  TF_DataType dtype = getTensorFlowDataType(type, SILFn.getLocation());
+  switch (dtype) {
+  case TF_FLOAT:
+    value = APFloat((float)1.0).bitcastToAPInt();
+    break;
+  case TF_DOUBLE:
+    value = APFloat((double)2.0).bitcastToAPInt();
+    break;
+  case TF_BOOL:
+    value = APInt(1, "0", 10);
+    break;
+  case TF_INT8:
+  case TF_UINT8:
+    value = APInt(8, "aa", 16);
+    break;
+  case TF_INT16:
+  case TF_UINT16:
+    value = APInt(16, "aaaa", 16);
+    break;
+  case TF_INT32:
+  case TF_UINT32:
+    value = APInt(32, "aaaaaa", 16);
+    break;
+  case TF_INT64:
+  case TF_UINT64:
+    value = APInt(64, "aaaaaaaa", 16);
+    break;
+  case TF_STRING:
+  case TF_COMPLEX64:  // Single-precision complex
+  case TF_QINT8:      // Quantized int8
+  case TF_QUINT8:     // Quantized uint8
+  case TF_QINT32:     // Quantized int32
+  case TF_BFLOAT16:   // Float32 truncated to 16 bits.  Only for cast ops.
+  case TF_QINT16:     // Quantized int16
+  case TF_QUINT16:    // Quantized uint16
+  case TF_COMPLEX128: // Double-precision complex
+  case TF_HALF:
+  case TF_RESOURCE:
+  case TF_VARIANT:
+    llvm_unreachable("Cannot deal with other undef node types at this point");
+  }
+
+  // Create a graph node
+  auto &graphFn = getCurrentGraphFunction();
+  std::string nodeName("UndefConst" + type.getAsString());
+  // escape the node name.
+  for (std::string::size_type i = 0; i < nodeName.size(); ++i) {
```

Comment:
it'd be good to unify this with escapeOpName() at https://github.com/bgogul/swift/blob/4cdd337fd50a80ab94e18544954ad8cd76857fcb/lib/SILOptimizer/Mandatory/TFLowerGraph.cpp#L627. Picking either one of them is fine for me.